flocker/acceptance/test_api.py

```diff
@@ -369,6 +369,61 @@ def can_connect():
         d.addCallback(check_result)
         return d
 
+    def test_create_container_with_environment(self):
+        """
+        Create a container including environment variables on a single-node
+        cluster.
+        """
+        data = {
+            u"name": "my_env_container",
+            u"host": None,
+            u"image": "busybox:latest",
+            u"environment": {u"ACCEPTANCE_ENV_LABEL": 'acceptance test ok'}
+        }
+        waiting_for_cluster = wait_for_cluster(test_case=self, node_count=1)
+
+        def create_container(cluster, data):
+            data[u"host"] = cluster.nodes[0].address
+            return cluster.create_container(data)
+
+        d = waiting_for_cluster.addCallback(create_container, data)
+
+        def check_result((cluster, response)):
+            self.assertEqual(response, data)
+
+        def inspect_container():
+            try:
+                remote_data = run_SSH(
+                    22, 'root', data[u"host"],
+                    [
+                        b"docker",
+                        b"inspect",
+                        b"flocker--{0}".format(data[u"name"])
+                    ], None
+                )
+            except Exception:
+                return False
+            container_data = loads(remote_data)
+            env_data = container_data[0]['Config']['Env']
+            result = dict()
+            for var in env_data:
+                try:
+                    label, value = var.split("=")
+                except ValueError:
+                    continue
+                result[label] = value
+            return result
+
+        def verify_environment(actual, expected):
+            for key, value in expected.iteritems():
+                self.assertEqual(actual[key], value)
```

Comment:
Assertions in a loop like this make for very difficult to debug test failures.  By doing an assertion in a loop, if one assertion fails you don't get to see the results of any later iterations (the normal problem with multiple assertions in one test).  And on top of that, the exception that gets raised doesn't tell you which key the problem was with - so you don't really know how the test failed, only that it did.

Also, in this case, it seems like there's only actually one item in `expected` anyway.  So it would probably be better to just make an assertion about that one specific item.  Alternatively, use one of the useful assertion helpers like `assertDictContainsSubset`.